src/resource/iri.internal.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { IriString } from "../interfaces";
+
+/**
+ * This function normalizes IRIs as managed by the server to ease accurate comparison.
+ * @param iri
+ * @hidden
+ */
+export function normalizeServerSideIri(iri: IriString): IriString {
```

Comment:
At the bottom of `acp/acp.ts`, this exact function also exists - might be a good moment to replace that with this one.

Also, would you mind adding this as an additional check to all the `save*` (and `overwrite*`) functions? I'm afraid similar bugs might be lurking in there - for example, line 244 of `solidDataset.ts` sets the URL of the returned SolidDataset to the URL passed to `saveSolidDatasetAt`. (Which I suppose would be better if it would take that from `response.url` anyway...)

Reply:
Good catch, I didn't realize it was there :) 400531cb49fb424dbfaea936aa91a535dd6b0dd4.

As for other `save` or `overwrite` methods, I'm not sure if anything is required: the bug in this case is that we use the IRI comparison to check whether we want to `PUT` or `PATCH`. When saving in container or overwriting, the method to use is explicit, so no IRI comparison is used, which prevents the bug from manifesting I think.

Reply:
Fair enough - I suppose it's only relevant in determining the source URL when returning it, fetching should be fine.